### PR TITLE
[cleanup] Remove unnecessary Map/Set module creation

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

**Review by herbelin — APPROVED** [review]
Looks good to me.

File: vernac/library.ml [hunk 1]
```diff
@@ -103,25 +103,21 @@ type library_summary = {
   libsum_digests : Safe_typing.vodigest;
 }
 
-module LibraryOrdered = DirPath
-module LibraryMap = Map.Make(LibraryOrdered)
-module LibraryFilenameMap = Map.Make(LibraryOrdered)
```
Comment (ppedrot) [inline vernac/library.ml]: This kind of redeclaration has a semantic intent, I am not sure it should be removed.

Comment (herbelin) [reply vernac/library.ml]: @ppedrot: I was actually about to tell something similar, that it has a "documentation" intent. So, maybe, a compromise would indeed be to keep this one.

Comment (SkySkimmer) [reply vernac/library.ml]: Functors are applicative so while they may be differentiated from DPmap by the mli these 1 are not different from each other though

Comment (ejgallego) [reply vernac/library.ml]: I did indeed wonder what to do about these, but in the end I found overall code clearer without the aliases, they don't seem to bring much to me (on the contrary they do obscure the code a bit), but no problem in keeping them if you find them useful.

**Review by SkySkimmer — APPROVED** [review]
Let's merge

File: ide/wg_Completion.ml [hunk 2]
```diff
@@ -8,17 +8,7 @@
...
-  type t = string
-  let compare s1 s2 =
-    (* we use first size, then usual comparison *)
-    let d = String.length s1 - String.length s2 in
-    if d <> 0 then d
-    else compare s1 s2
-end
-
-module Proposals = Set.Make(StringOrd)
+module Proposals = CString.Set
```
Comment (ppedrot) [inline ide/wg_Completion.ml]: Sorry to only wake up after the merge, but *as duly explained by the comment* this is wrong. Proposals need to be ordered by length first, for UI reasons.

Comment (ppedrot) [reply ide/wg_Completion.ml]: See #11965.

Comment (ejgallego) [reply ide/wg_Completion.ml]: Ups, sorry, fix merged.

**Pull request merged**